spec/system/ai_bot/artifact_spec.rb

```diff
@@ -48,30 +48,26 @@
     find(".ai-artifact__click-to-run button").click
 
     artifact_element_selector = ".ai-artifact[data-ai-artifact-id='#{ai_artifact.id}']"
-    iframe_selector = "#{artifact_element_selector} iframe"
+    artifact_element = find(artifact_element_selector)
 
-    expect(page).to have_css(iframe_selector)
-
-    iframe_element = find(iframe_selector)
-    expect(iframe_element["data-custom-message"]).to eq("hello-from-post")
-    expect(iframe_element["data-post-author-id"]).to eq(author.id.to_s)
+    expect(artifact_element).to have_css("iframe[data-custom-message='hello-from-post']")
+    expect(artifact_element).to have_css("iframe[data-post-author-id='#{author.id}']")
```

Comment:
You can't check data attributes on the element itself in capybara, so the best way is to do this.